task-2/src/components/MoviesList/index.js

```diff
@@ -1,32 +1,52 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
 
 import PropTypes from 'prop-types';
-import { useDispatch } from 'react-redux';
 
-import { movieData } from '../../core/data';
-import { switchBanner } from '../../store/actionCreators/actionCreators';
+import axios from 'axios';
+
+import { switchBanner, allLoadedMovies } from '../../store/actionCreators';
 
 import { Movie } from '../Movie';
 
 import './index.css';
 
+const movieApi = 'http://localhost:4000/movies?limit=10';
+
 export const MovieList = ({ show }) => {
+	const allMovies = useSelector((state) => state.allMovies.allMovies[0]);
+	// console.log(allMovies);
 	const dispatch = useDispatch();
-	const movies = movieData.map((movie) => (
+
+	const apiFetchMovies = async () => {
```

Comment:
Consider moving the request function to the above created "api" folder.
Do keep in mind that you don't need the dispatch part there, you can keep it in the component.

Reply:
fixed.